tools/convert/converters/converter-csv.cpp

```diff
@@ -0,0 +1,231 @@
+    size_t specific_attributes = 0;
+    if (val_in_range(_stream_type, { RS2_STREAM_GYRO,RS2_STREAM_ACCEL }))
+        specific_attributes = 3;
+    if (val_in_range(_stream_type, { RS2_STREAM_POSE }))
+        specific_attributes = 7;
+
+    for (auto i = 0; i < specific_attributes; i++)
+        ss << "," << _params[i];
+    ss << std::endl;
+    return ss.str().c_str();
+}
+
+converter_csv::converter_csv(const std::string& filePath, rs2_stream streamType)
+    : _filePath(filePath)
+    , _streamType(streamType)
+    , _imu_pose_collection()
+    , _sub_workers_joined(false)
+    , _m()
+    , _cv()
+{
+}
+
+void converter_csv::convert_depth(rs2::depth_frame& depthframe)
+{
+    if (frames_map_get_and_set(depthframe.get_profile().stream_type(), depthframe.get_frame_number())) {
+        return;
+    }
+
+    start_worker(
+        [this, &depthframe] {
+
+            std::stringstream filename;
+            filename << _filePath
+                << "_" << depthframe.get_profile().stream_name()
+                << "_" << std::setprecision(14) << std::fixed << depthframe.get_timestamp()
+                << ".csv";
+
+            std::stringstream metadata_file;
+            metadata_file << _filePath
+                << "_" << depthframe.get_profile().stream_name()
+                << "_metadata_" << std::setprecision(14) << std::fixed << depthframe.get_timestamp()
+                << ".txt";
+
+            std::string filenameS = filename.str();
+            std::string metadataS = metadata_file.str();
+
+            add_sub_worker(
+                [filenameS, metadataS, depthframe] {
+                    std::ofstream fs(filenameS, std::ios::trunc);
+
+                    if (fs) {
+                        for (int y = 0; y < depthframe.get_height(); y++) {
+                            auto delim = "";
+
+                            for (int x = 0; x < depthframe.get_width(); x++) {
+                                fs << delim << depthframe.get_distance(x, y);
+                                delim = ",";
+                            }
+                            fs << '\n';
+                        }
+                        fs.flush();
+                    }
+                    metadata_to_txtfile(depthframe, metadataS);
+                });
+            wait_sub_workers();
+        });
+}
+
+std::string converter_csv::get_time_string() const
+{
+    /*struct tm newtime;
+    time_t now = time(0);
+    localtime_s(&newtime, &now);
+
+    std::stringstream ss;
+    ss << newtime.tm_hour;
+    ss << newtime.tm_min;
+    ss << newtime.tm_sec;
+    return ss.str();*/
+    auto t = time(nullptr);
+    char buffer[20] = {};
+    const tm* time = localtime(&t);
+    if (nullptr != time)
+        strftime(buffer, sizeof(buffer), "%Y%m%d%H%M%S", time);
+
+    return std::string(buffer);
+}
+
+void converter_csv::output_to_csv()
+{
+    if (!_imu_pose_collection.size())
```

Comment:
IMO/Pose collection can be empty. Replace throw with user notification and exit gracefully

Reply:
let's talk about this please

Reply:
Done